### PR TITLE
Patch the ability to use a distro's own repo for the MariaDB package

### PR DESCRIPTION
**Pull request by zwall-bp**
# Description

There were converge issues when attempting to use either `mariadb_server_install` or `mariadb_client_install` with `setup_repo` set to `false`, allowing to use the distro's stock package repository. 

For Debian-based systems, it was still utilizing the `version` property, with the default value leading to an inconsistency to the current available package on debian's repo.

For RHEL-based systems, there was a naming issue where the client package is called `MariaDB-client` on MariaDB's own repository, but it's called `mariadb` on the RHEL repos. Another issue found was the case sensitivity of the package manager. While MariaDB's repo uses the convention `MariaDB-server`, stock RHEL uses `mariadb-server`.

## Issues Resolved

- Patched Debian based system's usage of the `version`. (The addition of using version alongside the stock repo wasn't considered, as there is usually only one version available at a time, each distro having a different version)
- Patched the ability to install the stock MariaDB client and server on DNF-based systems.

## Issues Witnessed

The original `server_install.rb` integration tests checks to see if  the `mysql` service exists. `amazonlinux-7`, `centos-7`, and `ubuntu-1804`'s stock packages do not create the `mysql` service. The `mariadb` service exists, however.

## Check List

- [ ] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [x] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## Unreleased
 
+## 5.2.10 - *2023-03-02*
```
Comment (ramereth) [inline CHANGELOG.md]: Do not add a release, our automated process will do this once it gets merged.

File: resources/client_install.rb [hunk 2]
```diff
@@ -18,6 +18,8 @@
 provides :mariadb_client_install
 unified_mode true
 
+include MariaDBCookbook::Helpers
+
```
Comment (ramereth) [inline resources/client_install.rb]: This is not needed, please remove.

**Review by ramereth — CHANGES_REQUESTED** [review]
Please update `.github/workflows/ci.yml` so that the two new suites will be included in the CI run.

**Comment by zwall-bp** [comment]
I still have an issue that needs to be patched. And I think I might have deleted the CHANGELOG file in my previous commit.

**Comment by ramereth** [comment]
@sectopodwreck it doesn't look like you rebased from the `origin/main` branch. Can you please do that?

**Comment by damacus** [comment]
Can you, undelete the CHANGELOG please

**Comment by zwall-bp** [comment]
There seems to be an issue in the converging of some Debian-based systems where `apply-mariadb-root-password` is not idempotent, even though the password is correctly set on the first pass through. I am unsure as to why the guard isn't working.

**Comment by zwall-bp** [comment]
There is an issue with the Amazon package servers right now, we might need to have to re-test sometime in the future.

Other than that, I am unable to add a label onto this project, if anyone else can please add an appropriate one, that can help with one of the failing tests.

With that, this PR should now help with ensuring that the `mariadb_server_install` resource will work perfectly, by default, when the developer sets `setup_repo` to `false`.

**Review by damacus — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [5.2.16](https://github.com/sous-chefs/mariadb/releases/tag/5.2.16)

**Pull request merged**